packages/dev/serializers/src/USD/USDSerializer.ts

```diff
@@ -0,0 +1,710 @@
+        const pad = "			";
+        const inputs = [];
+        const samplers = [];
+        const material = _material as PBRMaterial;
+        const scene = material.getScene();
+
+        const getChannelIndex = (channel: string) => {
+            switch (channel) {
+                case "r":
+                    return 0;
+                case "g":
+                    return 1;
+                case "b":
+                    return 2;
+                case "a":
+                    return 3;
+                default:
+                    return 0;
+            }
+        };
+
+        async function createDynamicTextureFromChannel(texture: Texture, channel: string) {
+            const dt = new DynamicTexture("dt", { width: texture.getSize().width, height: texture.getSize().height }, scene, false);
+            const ctx = dt.getContext();
+            const img = (await texture.readPixels()) as any;
+            const data = (ctx as any).createImageData(texture.getSize().width, texture.getSize().height);
+            const count = texture.getSize().width * texture.getSize().height * 4;
+
+            for (let i = 0; i < count; i += 4) {
+                const value = channel === "r" ? 255 - img[i + getChannelIndex(channel)] : img[i + getChannelIndex(channel)];
+                data.data[i] = value;
+                data.data[i + 1] = value;
+                data.data[i + 2] = value;
+                data.data[i + 3] = 255;
+            }
+
+            ctx.putImageData(data, 0, 0);
+            dt.update(texture.invertY);
+            return dt;
+        }
+
+        async function breakApartMetallicRoughnessAo(texture: Texture) {
+            const ao = await createDynamicTextureFromChannel(texture, "r");
+            ao.name = "ao";
+            const roughness = await createDynamicTextureFromChannel(texture, "g");
+            roughness.name = "roughness";
+            const metallic = await createDynamicTextureFromChannel(texture, "b");
+            metallic.name = "metallic";
+            return { ao, roughness, metallic };
+        }
+
+        function buildTexture(texture: Texture, mapType: string, color: Color3 | undefined = undefined) {
+            const id = texture.uniqueId + "_" + texture.invertY;
+            textures[id] = texture;
+
+            const uv = texture.coordinatesIndex > 0 ? `st${texture.coordinatesIndex}` : `st`;
+
+            /* TODO CHECK THIS */
+            const WRAPPINGS = [
+                "clamp", // ClampToEdgeWrapping
+                "repeat", // RepeatWrapping
+                "mirror", // MirroredRepeatWrapping
+            ];
+
+            const repeat = new Vector2(texture.wrapU === Texture.CLAMP_ADDRESSMODE ? 0 : 1, texture.wrapV === Texture.CLAMP_ADDRESSMODE ? 0 : 1);
+            const offset = new Vector2(texture.uOffset, texture.vOffset);
+            const rotation = 0; /*TODO, remember what value this is*/

```

Comment:
This is probably `texture.wAng` => you should test it and make sure the rendering when converted to udsz matches the source.

Reply:
Ill have to check this out sometime!